libcudacxx/include/cuda/__memory/get_device_address.h

```diff
@@ -36,19 +38,41 @@ _CCCL_BEGIN_NAMESPACE_CUDA
 //! @param __device_object the object residing in device memory
 //! @return Valid pointer to the device object
 template <class _Tp>
-[[nodiscard]] _CCCL_API inline _Tp* get_device_address(_Tp& __device_object)
+[[nodiscard]] CCCL_DEPRECATED _CCCL_API inline _Tp* get_device_address(_Tp& __device_object)
```

Comment:
I am not to big of a fan of deprecating this. A lof of users are still in a single device stup and we are making life much harder for them. I would love to have an additional doc string at the beginning

```
//! @warn The user must ensure that the current device is properly set to the device the object was allocated on
```